enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -157,16 +160,41 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                 Utils.setTag(jwtAuthenticatorInfoSpan, APIConstants.LOG_TRACE_ID,
                         ThreadContext.get(APIConstants.LOG_TRACE_ID));
             }
-            String jwtToken = retrieveAuthHeaderValue(requestContext);
-
-            if (jwtToken == null || !jwtToken.toLowerCase().contains(JWTConstants.BEARER)) {
+            String authHeaderVal = retrieveAuthHeaderValue(requestContext);
+            if (authHeaderVal == null || !authHeaderVal.toLowerCase().contains(JWTConstants.BEARER)) {
                 throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
                         APISecurityConstants.API_AUTH_MISSING_CREDENTIALS, "Missing Credentials");
             }
-            String[] splitToken = jwtToken.split("\\s");
+            String[] splitToken = authHeaderVal.split("\\s");
+            String token = authHeaderVal;
             // Extract the token when it is sent as bearer token. i.e Authorization: Bearer <token>
             if (splitToken.length > 1) {
-                jwtToken = splitToken[1];
+                token = splitToken[1];
+            }
+            // Handle PAT logic
+            if (token.startsWith(APIKeyConstants.PAT_PREFIX)) {
```

Comment:
shall we move this to a private function

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/3592/commits/8e7a16edf9d71b7676631480e5824113ff5b30de